### PR TITLE
Refactor Subtasks for Homepage + Add Planning Review

### PR DESCRIPTION
**Pull request by dhaselhan**

### CODE REVIEW

File: alcs-frontend/src/app/features/board/board.component.ts [hunk 1]
```diff
@@ -105,6 +106,59 @@ export class BoardComponent implements OnInit, OnDestroy {
     this.destroy.complete();
   }
 
+  async onSelected(card: CardSelectedEvent) {
+    switch (card.cardType) {
```
Comment (dhaselhan) [inline alcs-frontend/src/app/features/board/board.component.ts:110]: I should move setUrl to here and then I can delete it from each function below.

File: alcs-api/src/application/application.service.ts [hunk 2]
```diff
@@ -199,7 +199,7 @@ export class ApplicationService {
     return application;
   }
 
-  async getAllApplicationsWithIncompleteSubtasks(subtaskType: string) {
+  async getBySubtaskType(subtaskType: string) {
```
Comment (mhuseinov) [inline alcs-api/src/application/application.service.ts:202]: It still uses the incomplete filter, I think it should be reflected somehow

File: alcs-api/src/planning-review/planning-review.service.ts [hunk 3]
```diff
@@ -108,4 +108,27 @@ export class PlanningReviewService {
       relations: this.DEFAULT_RELATIONS,
     });
   }
+
+  async getBySubtaskType(subtaskType: string) {
```
Comment (mhuseinov) [inline alcs-api/src/planning-review/planning-review.service.ts:112]: the same about incomplete filter

File: alcs-frontend/src/app/features/board/board.component.ts [hunk 4]
```diff
@@ -105,6 +106,60 @@ export class BoardComponent implements OnInit, OnDestroy {
...
+        await this.openAppCardDetailDialog(card.uuid);
+        break;
+      case CardType.RECON:
+        await this.openReconCardDetailDialog(card.uuid);
+        break;
+      case CardType.PLAN:
+        await this.openPlanningCardDialog(card.uuid);
+        break;
+      default:
+        console.error('Card type is not configured for a dialog');
```
Comment (mhuseinov) [inline alcs-frontend/src/app/features/board/board.component.ts:122]: Should we show an error toast to user "Failed to open card". I know that this should not happen but if we already providing a error message here

File: alcs-api/src/application/application.service.ts (hunk 2 above)
Comment (dhaselhan) [reply alcs-api/src/application/application.service.ts]: Renamed

File: alcs-api/src/planning-review/planning-review.service.ts (hunk 3 above)
Comment (dhaselhan) [reply alcs-api/src/planning-review/planning-review.service.ts]: Renamed

File: alcs-frontend/src/app/features/board/board.component.ts (hunk 4 above)
Comment (dhaselhan) [reply alcs-frontend/src/app/features/board/board.component.ts:122]: Done!

**Pull request merged**